pyrorl/pyrorl/envs/map_helpers/create_map_info.py

```diff
@@ -0,0 +1,210 @@
+            orientation = random.choice(["north", "south", "east", "west"])
+
+            done = False
+            while not done:
+
+                # we want to make sure that the current path will not intersect with itself
+                direction_chosen = False
+                num_steps = random.randint(2, 4)
+
+                while not direction_chosen:
+                    # have a bias toward going straight
+                    direction_index = random.randint(0, 1)
+                    if direction_index == 2:
+                        direction_index = random.randint(1, 2)
+                    direction = directions[direction_index]
+
+                    if orientation == "north" and direction == "left":
+                        if cur_row - num_steps < x_min:
+                            direction_chosen = True
+                    elif orientation == "south" and direction == "right":
+                        if cur_row + num_steps > x_max:
+                            direction_chosen = True
+                    elif orientation == "east" and direction == "left":
+                        if cur_col + num_steps > y_max:
+                            direction_chosen = True
+                    elif orientation == "west" and direction == "right":
+                        if cur_col - num_steps < y_min:
+                            direction_chosen = True
+                    else:
+                        direction_chosen = True
+
+                row_update = orientations[orientation][direction][0][0]
+                col_update = orientations[orientation][direction][0][1]
+
+                for _ in range(num_steps):
+                    cur_row += row_update
+                    cur_col += col_update
+
+                    if cur_row > x_max:
+                        x_max = cur_row
+                    if cur_row < x_min:
+                        x_min = cur_row
+                    if cur_col > y_max:
+                        y_max = cur_col
+                    if cur_col < y_min:
+                        y_min = cur_col
+
+                    # the population center is on the edge of the map, so we don't want to add a path in this direction
+                    if (
+                        cur_row == -1
+                        or cur_row == num_cols
+                        or cur_col == -1
+                        or cur_col == num_rows
+                    ):
+                        done = True
+                        break
+
+                    current_path.append([cur_row, cur_col])
+                    if (
+                        cur_row == 0
+                        or cur_row == num_cols - 1
+                        or cur_col == 0
+                        or cur_col == num_rows - 1
+                    ):
+                        done = True
+                        paths.append(current_path)
+                        paths_to_pops[path_num] = [[pop_row, pop_col]]
```

Comment:
Should this be an append here? Is this line implying that for each path, it only has one populated area that can use it?

Reply:
That is how the map generation is structured right now

Reply:
How long would it take to change/make more general?